pina/model/fno.py

```diff
@@ -1,54 +1,71 @@
+                 output_numb_fields,
+                 n_modes,
+                 dimensions=3,
+                 padding=8,
+                 padding_type="constant",
+                 inner_size=20,
+                 n_layers=2,
+                 func=nn.Tanh,
+                 layers=None):
+        """
+        Implementation of Fourier Integral Kernel network.
+
+        This class implements the Fourier Integral Kernel network, which is a
+        PINA implementation of Fourier Neural Operator kernel network.
+        It performs global convolution by operating in the Fourier space.
+
+        .. seealso::
 
+            **Original reference**: Li, Z., Kovachki, N., Azizzadenesheli, K., Liu, B.,
+            Bhattacharya, K., Stuart, A., & Anandkumar, A. (2020). *Fourier neural operator for
+            parametric partial differential equations*.
+            DOI: `arXiv preprint arXiv:2010.08895.
+            <https://arxiv.org/abs/2010.08895>`_
 
-class FNO(torch.nn.Module):
-    """
-    The PINA implementation of Fourier Neural Operator network.
-
-    Fourier Neural Operator (FNO) is a general architecture for learning Operators.
-    Unlike traditional machine learning methods FNO is designed to map
-    entire functions to other functions. It can be trained both with
-    Supervised learning strategies. FNO does global convolution by performing the
-    operation on the Fourier space.
-
-    .. seealso::
-
-        **Original reference**: Li, Z., Kovachki, N., Azizzadenesheli, K., Liu, B.,
-        Bhattacharya, K., Stuart, A., & Anandkumar, A. (2020). *Fourier neural operator for
-        parametric partial differential equations*.
-        DOI: `arXiv preprint arXiv:2010.08895.
-        <https://arxiv.org/abs/2010.08895>`_
-    """
-
-    def __init__(
-        self,
-        lifting_net,
-        projecting_net,
-        n_modes,
-        dimensions=3,
-        padding=8,
-        padding_type="constant",
-        inner_size=20,
-        n_layers=2,
-        func=nn.Tanh,
-        layers=None,
-    ):
+        :param input_numb_fields: Number of input fields.
+        :type input_numb_fields: int
+        :param output_numb_fields: Number of output fields.
+        :type output_numb_fields: int
+        :param n_modes: Number of modes.
+        :type n_modes: int or list[int]
+        :param dimensions: Number of dimensions (1, 2, or 3).
+        :type dimensions: int
+        :param padding: Padding size, defaults to 8.
+        :type padding: int
+        :param padding_type: Type of padding, defaults to "constant".
+        :type padding_type: str
+        :param inner_size: Inner size, defaults to 20.
+        :type inner_size: int
+        :param n_layers: Number of layers, defaults to 2.
+        :type n_layers: int
+        :param func: Activation function, defaults to nn.Tanh.
+        :type func: torch.nn.Module
+        :param layers: List of layer sizes, defaults to None.
+        :type layers: list[int]
```

Comment:
for primitive type, use the compact form

Reply:
done!